core/chains/evm/chain.go

```diff
@@ -110,12 +110,14 @@ func newChain(dbchain types.Chain, nodes []types.Node, opts ChainSetOpts) (*chai
 		headTracker = opts.GenHeadTracker(dbchain, headBroadcaster)
 	}
 
+	logPoller := logpoller.NewLogPoller(logpoller.NewORM(chainID, db, l, cfg), client, l, cfg.EvmLogPollInterval(), int64(cfg.EvmFinalityDepth()), int64(cfg.EvmLogBackfillBatchSize()))
```

Comment:
Technically out of scope, but this constructor seems to expect the callers to do a lot of duplicate work that it could do internally itself, no? i.e. `NewLogPoller(db, client, cfg, l)`

Reply:
yeah it could define its own interface with just those handful of methods and accept that, but it just makes testing harder with no obvious gain imo